en/dev_log/ulog_file_format.md

```diff
@@ -137,51 +142,57 @@ It is possible that there are more fields appended at the end of this message in
+- A type can be used before it's defined.
+  - e.g. The message `MessageA:MessageB[2] msg_b` can come before the `MessageB:uint_8[3] data`
+- There can be arbitrary nesting but **no circular dependencies**
+  - e.g. `MessageA:MessageB[2] msg_b` & `MessageB:MessageA[4] msg_a`
 
 Some field names are special:
-- `timestamp`: every [Subscription Message](#a--subscription-message) must include a timestamp field
+- `timestamp`: every [Subscription Message](#a-subscription-message) must include a timestamp field
   - Its type can be: `uint64_t` (currently the only one used), `uint32_t`, `uint16_t` or `uint8_t`.
-  - The unit is always microseconds, except for in `uint8_t` it's milliseconds.
+  - The unit is always microseconds, except for in `uint8_t` where the unit is in milliseconds.
   - A log writer must make sure to log messages often enough to be able to detect wrap-arounds and a log reader must handle wrap-arounds (and take into account dropouts).
   - The timestamp must always be monotonic increasing for a message series with the same `msg_id`.
-- Padding: field names that start with `_padding` should not be displayed and their data must be ignored by a reader.
+- `_padding{}`: field names that start with `_padding` (e.g. `_padding[3]`) should not be displayed and their data must be ignored by a reader.
   - These fields can be inserted by a writer to ensure correct alignment.
   - If the padding field is the last field, then this field will not be logged, to avoid writing unnecessary data.
   - This means the `message_data_s.data` will be shorter by the size of the padding.
   - However the padding is still needed when the message is used in a nested definition.
 
 #### 'I': Information Message
 
+The Information message defines a dictionary type definition `key` : `value` pair for any information, including but not limited to Hardware version, Software version, Build toolchain for the software, etc.
+
 ```c
-struct message_info_s {
-  struct message_header_s header;
+struct ulog_message_info_header_s {
+  uint16_t msg_size;
+  uint8_t msg_type = 'I';
   uint8_t key_len;
-  char key[key_len];
-  char value[header.msg_size-1-key_len]
+  char key[255];
 };
 ```
-* `key` is a plain string, as in the format message (can also be a custom type), but consists of only a single field without ending `;`, eg. `float[3] myvalues`
-* `value` contains the data as described by `key`
+
+- `key_len`: Length of the key value
+- `key`: String with the `key` and `value` information combined: e.g. `char[5] sys_toolchain_ver9.4.0`
+  - The `key` part has the format identical as the field in the [Format message](#f-format-message) (can also be a custom type), but consists of only a single field without ending `;`, eg. `char[5] sys_toolchain_ver`
+  - The `value` contains the data corresponding to the `key`, and is written right after the `key`, e.g. `9.4.0`.
```

Comment:
It was very confusing to explain how the key string of the struct differs from the concept of key : value of the information we are encompassing inside the key string.

Maybe worth changing the naming in the PX4's side to clarify what this field actually does? @bkueng

Reply:
PR here : https://github.com/PX4/PX4-Autopilot/pull/19675